api/views.py

```diff
@@ -362,3 +364,47 @@ def destroy(self, request, *args, **kwargs):
         self.perform_destroy(instance)
         data = {"detail": "Deleted Successfully"}
         return Response(data=data, status=status.HTTP_204_NO_CONTENT)
+
+
+class AssetsImportViewSet(APIView):
+    parser_classes = (MultiPartParser,)
+
+    def post(self, request):
+        file_obj = request.data['file']
+        model_number = AssetModelNumber(
+            model_number=request.data['model_number'])
+
+        kwargs = {}
+
+        year_of_manufacture = request.data.get('year_of_manufacture')
+        processor_speed = request.data.get('processor_speed')
+        screen_size = request.data.get('screen_size')
+        processor_type = request.data.get('processor_type')
+        storage = request.data.get('storage')
+        memory = request.data.get('memory')
+
+        kwargs.update({'year_of_manufacture': year_of_manufacture,
+                       'processor_speed': processor_speed,
+                       'screen_size': screen_size,
+                       'processor_type': processor_type,
+                       'storage': storage,
+                       'memory': memory})
+
+        asset_specs = AssetSpecs.objects.get_or_create(**kwargs)
+
+        assets = []
+
+        for pos, line in enumerate(file_obj):
```

Comment:
i think it might be better to use the csv module which takes care of column headers. if a file has the columns ordered differently the implementation here would be wrong. eg if the imported file has serial number in the first column.

Reply:
Noted